R/jjcsformats.R

```diff
@@ -1,88 +1,89 @@
-format_xx_fct <- function(roundmethod = c("sas", "iec"), na_str_dflt = "NE",
-                          replace_na_dflt = TRUE) {
-  roundmethod <- match.arg(roundmethod)
-
-  if (roundmethod == "sas") {
-    roundfunc <- tidytlg::roundSAS
-  } else {
-    roundfunc <- round
-  }
-
-  fnct <- function(str, na_str = na_str_dflt) {
-    if (grepl("xxx.", str, fixed = TRUE)) {
-      stop(
-        "Error: jjcs_format_xx: do not use xxx. in input str, replace by xx. instead."
-      )
+#' value <- c(1.65, 8.645)
+#' fmt <- jjcsformat_xx("xx.x")
+#' is.function(fmt)
+#' fmt
+#' format_value(value[1], fmt, round_type = "sas")
+#' format_value(value[1], fmt, round_type = "iec")
+#' fmt(value[1])
+#' 
+#' fmt2 <- jjcsformat_xx("xx.x (xx.xxx)")
+#' is.function(fmt2)
+#' value <- c(1.65, 8.645)
+#' format_value(value, fmt2, round_type = "sas")
+#' format_value(value, fmt2, round_type = "iec") 
+#' #only possible when resulting format is a function
+#' fmt2(value, round_type = "sas")
+#' 
+#' value <- c(1.65, NA)
+#' format_value(value, fmt2, round_type = "iec", na_str = c("ne1", "ne2"))
+#' fmt2(value, round_type = "iec", na_str = c("ne1", "ne2"))
+
+
+jjcsformat_xx <- function(str,
+    na_str = na_str_dflt,
+    na_str_dflt = "NE",
+    replace_na_dflt = TRUE) {
+
+      if (grepl("xxx.", str, fixed = TRUE)) {
+      stop("Error: jjcsformat_xx do not use xxx. in input str, replace by xx. instead.")
     }
+    
+    if (is_valid_format(str)) {
+      rtable_format <- str
+    } else { 
+    
     if (!grepl("xx", str, fixed = TRUE)) {
-      stop("Error: jjcs_format_xx: input str must contain xx.")
+      stop("Error: jjcsformat_xx input str must contain xx.")
     }
-    positions <- gregexpr(
-      pattern = "xx\\.?x*",
-      text = str,
-      perl = TRUE
-    )
+    positions <- gregexpr(pattern = "xx\\.?x*",
+                          text = str,
+                          perl = TRUE)
     x_positions <- regmatches(x = str, m = positions)[[1]]
-    ### str is splitted into pieces as xx. xx xx.xxx
-    ### xx is no rounding
-    ### xx. rounding to integer
-    ### xx.x rounding to 1 decimal, etc
-
-    no_round <- function(x, na_str = na_str_dflt) {
-      if (is.na(x)) {
-        return(na_str)
-      } else {
-        return(x)
-      }
-    }
-
-    roundings <- lapply(X = x_positions, function(x) {
-      y <- strsplit(split = "\\.", x = x)[[1]]
-      ### "xx.x" will result in c("xx","x")
-      ### "xx." will result in "xx"
-      ### "xx" will remain "xx"
-
-      if (x == "xx") {
-        rounding <- no_round
-      } else {
-        rounding <- function(x, na_str = na_str_dflt) {
-          if (is.na(x)) {
-            return(na_str)
-          } else {
-            format(
-              roundfunc(x, digits = ifelse(length(y) > 1, nchar(y[2]), 0)),
-              nsmall = ifelse(length(y) > 1, nchar(y[2]), 0)
-            )
-          }
+    
+    roundings <- lapply(X = x_positions, function(fmt) {
+      rounding <- function(x,
```

Comment:
why are we defining the `rounding` function within the `lapply` anonymous function? Are we using closure shenanigans here? If so, do we really need to? And even if so it can probably be outside of the lapply and just in the surrounding function, right?